src/Documentation/Grain-Versioning/Grain-versioning.md

```diff
@@ -0,0 +1,35 @@
+# Grain versioning
+
+> [!WARNING]
+> This page describes how to use grain interface versioning. The versioning of
+> Grain state it out of the scope.
+
+## Overview
+On a given cluster, silos can support different versions of a grain type.
+![Cluster with different versions of a grain](version.png)
+In this example the client and Silo{1,2,3} were compiled with grain interface `A` version 1. Silo 4 was compiled with `A` version 2.
+
+## Limitations:
+-	No versioning on stateless worker
+-	Streaming agents are not versioned
```

Comment:
Maybe elaborate a little more on "Streaming agents are not versioned". Do you mean the public streaming interface, like IObserver, IObservable does not support versioning?  or streaming infrastructure doesn't support rolling upgrading, so if Orleans streaming has a breaking change in between two versions, and doing a rolling upgrade from Orleans v1 to v2, you will be facing failures?